core/services/keeper/registry_synchronizer_process_logs.go

```diff
@@ -309,3 +311,86 @@ func (rs *RegistrySynchronizer) handleUpkeepMigrated(broadcast log.Broadcast) {
 		rs.logger.Error(errors.Wrapf(err, "unable to mark KeeperRegistryUpkeepMigrated log as consumed,  log: %v", broadcast.String()))
 	}
 }
+
+func (rs *RegistrySynchronizer) handleUpkeepPausedLogs(done func()) {
+	defer done()
+	for {
+		broadcast, exists := rs.mailRoom.mbUpkeepPaused.Retrieve()
+		if !exists {
+			return
+		}
+		rs.handleUpkeepPaused(broadcast)
+	}
+}
+
+func (rs *RegistrySynchronizer) handleUpkeepPaused(broadcast log.Broadcast) {
+	txHash := broadcast.RawLog().TxHash.Hex()
+	rs.logger.Debugw("processing UpkeepPaused log", "txHash", txHash)
+	was, err := rs.logBroadcaster.WasAlreadyConsumed(broadcast)
+	if err != nil {
+		rs.logger.Error(errors.Wrap(err, "unable to check if log was consumed"))
+		return
+	}
+	if was {
+		return
+	}
+
+	pausedUpkeepId, err := rs.registryWrapper.GetUpkeepIdFromUpkeepPausedLog(broadcast)
+	if err != nil {
+		rs.logger.Error(errors.Wrap(err, "Unable to fetch upkeep ID from upkeep paused log"))
+		return
+	}
+
+	affected, err := rs.orm.BatchDeleteUpkeepsForJob(rs.job.ID, []utils.Big{*utils.NewBig(pausedUpkeepId)})
+	if err != nil {
+		rs.logger.Error(errors.Wrap(err, "unable to batch delete upkeeps"))
+		return
+	}
+	rs.logger.Debugw(fmt.Sprintf("paused %v upkeep registrations", affected), "txHash", txHash)
+
+	if err := rs.logBroadcaster.MarkConsumed(broadcast); err != nil {
+		rs.logger.Error(errors.Wrapf(err, "unable to mark KeeperRegistryUpkeepPaused log as consumed,  log: %v", broadcast.String()))
+	}
+}
+
+func (rs *RegistrySynchronizer) handleUpkeepCheckDataUpdatedLogs(done func()) {
+	defer done()
+	for {
+		broadcast, exists := rs.mailRoom.mbUpkeepCheckDataUpdated.Retrieve()
+		if !exists {
+			return
+		}
+		rs.handleUpkeepCheckDataUpdated(broadcast)
+	}
+}
+
+func (rs *RegistrySynchronizer) handleUpkeepCheckDataUpdated(broadcast log.Broadcast) {
+	txHash := broadcast.RawLog().TxHash.Hex()
+	rs.logger.Debugw("processing Upkeep check data updated log", "txHash", txHash)
+	was, err := rs.logBroadcaster.WasAlreadyConsumed(broadcast)
+	if err != nil {
+		rs.logger.Error(errors.Wrap(err, "unable to check if log was consumed"))
+		return
+	}
+	if was {
+		return
+	}
+
+	updateLog, err := rs.registryWrapper.ParseUpkeepCheckDataUpdatedLog(broadcast)
+	if err != nil {
+		rs.logger.Error(errors.Wrap(err, "Unable to parse update log from upkeep check data updated log"))
+		return
+	}
+
+	err = rs.orm.UpdateCheckData(utils.NewBig(updateLog.UpkeepID), rs.job.ID, updateLog.NewCheckData)
+	if err != nil {
+		rs.logger.Error(errors.Wrapf(err, "unable to update check data for upkeep %s", updateLog.UpkeepID.String()))
+		return
+	}
+
+	rs.logger.Debugw(fmt.Sprintf("updated check data with %T", updateLog), "txHash", txHash)
```

Comment:
Can we not log the full updateLog here? only upkeep ID would be better, can discuss reasoning on slack

Reply:
Done